runtime/ftplugin/qf.vim

```diff
@@ -14,3 +14,12 @@ let b:undo_ftplugin = "set stl<"
 
 " Display the command that produced the list in the quickfix window:
 setlocal stl=%t%{exists('w:quickfix_title')?\ '\ '.w:quickfix_title\ :\ ''}\ %=%-15(%l,%c%V%)\ %P
+
+autocmd Syntax <buffer>
```

Comment:
When the location list is updated, the window is reused, but the buffer isn't.  `filetype` gets set again and it starts over.